### ISSUE TITLE
panic: runtime error: makeslice: cap out of range

### ISSUE THREAD

**Issue by vsliouniaev** [jacksontj/promxy#251]
Getting a panic on promxy v0.0.56:

```
time="2019-12-02T17:08:23Z" level=info msg="promxy starting"
panic: runtime error: makeslice: cap out of range

goroutine 288 [running]:
github.com/jacksontj/promxy/vendor/github.com/prometheus/prometheus/promql.(*VectorSelector).String(0xc000e28000, 0xc000b24000, 0x26dc0e0)
	/go/src/github.com/jacksontj/promxy/vendor/github.com/prometheus/prometheus/promql/printer.go:180 +0x64
github.com/jacksontj/promxy/pkg/promclient.(*AddLabelClient).Query(0xc000aedd00, 0x2734200, 0xc000b24000, 0xc000dbe090, 0x21, 0x1bc69880, 0xed5773c33, 0x3937c20, 0x3, 0x3, ...)
	/go/src/github.com/jacksontj/promxy/pkg/promclient/label.go:114 +0x1b6
github.com/jacksontj/promxy/pkg/promclient.(*MultiAPI).Query.func1(0x2734200, 0xc000b24000, 0xc000e30e60, 0x0, 0xc00003a240, 0x2744c80, 0xc000aedd00, 0xc000dbe090, 0x21, 0x1bc69880, ...)
	/go/src/github.com/jacksontj/promxy/pkg/promclient/multi_api.go:289 +0xd2
created by github.com/jacksontj/promxy/pkg/promclient.(*MultiAPI).Query
	/go/src/github.com/jacksontj/promxy/pkg/promclient/multi_api.go:287 +0x300
```

**Comment by jacksontj** [comment]
@vsliouniaev do you have a query to reproduce this query?

**Comment by vsliouniaev** [comment]
Sure, I'm trying to localize this down a bit further:
- Seems like this is affecting older versions of promxy as well
- Query I'm using is `{prometheus="hello-metrics-test"}`, which is an external label that I'm adding to metrics I'm scraping from a prometheus like this:
```yaml
promxy:
  server_groups:
  - labels:
      cluster_id: my-test-cluster
      cluster_profile: dev
      cluster_region: cloud
      prometheus: hello-metrics-test
    relabel_configs:
    - source_labels: [__meta_kubernetes_pod_name, __meta_kubernetes_pod_label_app,
        __meta_kubernetes_pod_container_name, __meta_kubernetes_pod_container_port_name,
        __meta_kubernetes_pod_ready]
      regex: prometheus-prometheus-\d;prometheus;prometheus;web;true
      action: keep
    kubernetes_sd_configs:
    - role: pod
      namespaces:
        names:
        - hello-metrics-test
    anti_affinity: 30s
```

**Comment by jacksontj** [comment]
I can reproduce this error and have a fix in #252

**Comment by vsliouniaev** [comment]
Looks like a reasonable solution 👍
